src/D2L.CodeStyle.Analyzers/Immutability/ExplicitImmutableAttributeTransitivityAnalyzer.cs

```diff
@@ -106,10 +108,21 @@ CompilationStartAnalysisContext context
 					continue;
 				}
 
-				if( !hasTheImmutableAttribute( typeSymbol ) ) {
+				bool baseTypeIsMarkedImmutable = hasTheImmutableAttribute( typeSymbol );
+				bool baseTypeIsMarkedConditionallyImmutable = Attributes.Objects.ConditionallyImmutable.IsDefined( typeSymbol );
+
+				if( !baseTypeIsMarkedImmutable
+					&& !baseTypeIsMarkedConditionallyImmutable
+				) {
 					continue;
 				}
 
+				if( baseTypeIsMarkedConditionallyImmutable
+					&& typeIsMarkedConditionallyImmutable
+				) {
+					continue;
+				} 

```

Comment:
```csharp
			if( typeHasImmutableAttr ) {
				// Nothing to do: we already have the [Immutable] attribute.
				return;
			}
```

I left this in above, allowing for an immutable class to implement a conditionally immutable thing

```csharp
[ConditionallyImmutable] IFoo<[ConditionallyImmutable.OnlyIf] T> {}

[Immutable] class FullyImmutableFoo<T> : IFoo<T> {}
```

Reply:
Consider a "conflict" diagnostic for if we have `[Immutable]` and `[ConditionallyImmutable]` (doesn't have to be in this PR)

Reply:
Ah yeah, meant to add that to the to-do list